### PR TITLE
fix: verify author of each individual commit

### PR DESCRIPTION
**Pull request by ewanharris**
The GITHUB_USER environment variable does not always refer to the author of a PR, if that PR has to be approved before being ran, or the CLA check has been reran then it will point to the user who triggered that. By retrieving all commits and then inspecting each on this is a more correct way to check the validity

You can see some examples of this [here](https://github.com/tidev/titanium_mobile/runs/6342018068?check_suite_focus=true) (authorized by miga), and [here](https://github.com/tidev/vscode-titanium/runs/6313619138?check_suite_focus=true) (CLA check was reran by me)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,16 +17,28 @@ const body = await new Promise((resolve, reject) => {
...
+	const userHasSigned = signedUsers.includes(login);
+	if (userHasSigned) {
+		console.log(`User ${user} for commit ${sha} is authorized`);
+	} else {
+		console.log(`User ${user} for commit ${sha} not authorized`);
+
+	}
+});
 
-if (user === 'dependabot[bot]' || signed.find(u => u === user)) {
```
Comment (cb1kenobi) [inline index.js:25]: What about dependabot? Dependabot is not in the signedUsers array.

Comment (ewanharris) [reply index.js:25]: My bad, forgot to copy this across

**Comment by ewanharris** [comment]
One thing I'm not sure about here is as I'm introducing dependencies does this change how the action is used in repos? From my experience with other actions most of them create index.js with a tool like ncc

**Comment by cb1kenobi** [comment]
> One thing I'm not sure about here is as I'm introducing dependencies does this change how the action is used in repos?

I’m not sure. Are there any other actions we could see what they do? We could publish 1.1.0 and tag it, then update a single repo with the new tag.

If that works, then force redefine the v1 tag.

**Comment by ewanharris** [comment]
From looking around it's either check in node_modules or add a build step. I'd say the latter is preferable to me but both have tradeoffs

**Comment by cb1kenobi** [comment]
Yeah, adding node_modules to the repo sounds like a bad idea... or is it?

File: index.js [hunk 2]
```diff
@@ -15,16 +17,30 @@ const body = await new Promise((resolve, reject) => {
...
 });
 
-const signed = body
+const signedUsers = [...body
 	.match(/^\|.+\|$/mg)
 	?.slice(2)
-	.map(u => u.split('|')[2].trim());
+	.map(u => u.split('|')[2].trim()),
+	'dependabot[bot]'
+];
```
Comment (cb1kenobi) [inline index.js:25]: Couldn't you just done it the old school way and pushed dependabot onto the array? :)

**Review by cb1kenobi — APPROVED** [review]
LGTM! **APPROVED**

**Pull request merged**